cfme/automate/explorer.py

```diff
@@ -335,25 +335,30 @@ def make_path(cls, *names, **kwargs):
             #is equivalent to:
                 n = Namespace(name="bar", parent=Namespace(name="foo"))
         """
-
         domain = kwargs.get('domain', None)
+        create_on_init = kwargs.get('create_on_init', False)
 
         if len(names) == 1 and domain:
             names = list(names)
-            return cls(name=names.pop(), parent=domain)
+            return cls(name=names.pop(), parent=domain, create_on_init=create_on_init)
         elif names:
             names = list(names)
             if domain:
-                return cls(name=names.pop(), parent=cls.make_path(domain=domain, *names))
+                return cls(name=names.pop(), parent=cls.make_path(domain=domain,
+                    create_on_init=create_on_init, *names), create_on_init=create_on_init)
             else:
-                return cls(name=names.pop(), parent=cls.make_path(*names))
+                return cls(name=names.pop(), parent=cls.make_path(
+                    create_on_init=create_on_init, *names), create_on_init=create_on_init)
         else:
             return None
 
-    def __init__(self, name=None, description=None, parent=None, domain=None):
+    def __init__(self, name=None, description=None, parent=None, domain=None, create_on_init=False):
         self.name = name
         self.description = description
         self.parent = parent or (domain if isinstance(domain, Domain) else Domain.default)
+        if create_on_init:
```

Comment:
I appreciate that this is optional, but even still I'm averse to doing more than initialization in `__init__`. Is this strictly necessary, or can it go?